Archives/ClmFile.cpp

```diff
@@ -180,15 +135,15 @@ namespace Archives
 	{
 		headerOut.riffHeader.riffTag = RIFF;
 		headerOut.riffHeader.waveTag = WAVE;
-		headerOut.riffHeader.chunkSize = sizeof(headerOut.formatChunk) + 12 + m_IndexEntry[fileIndex].dataLength;
+		headerOut.riffHeader.chunkSize = sizeof(headerOut.riffHeader.waveTag) + sizeof(FormatChunk) + sizeof(ChunkHeader) + indexEntries[fileIndex].dataLength;
```

Comment:
This is a bit hard to read.

Initially it seemed like it sums sizes out of the order of appearance according to the file format, though it does actually match. The structs were declared in the order they should likely appear: RiffHeader, FormatChunk, ChunkHeader (for data), followed by the data.

It seems like RiffHeader and FormatChunk have hidden embedded ChunkHeader fields. The first 8 bytes seem to be a ChunkHeader, which contains the contents of the RIFF file. Hence the riffHeader chunk size seems to be everything after the first 8 byte header.

I believe the structs were modelled after the documentation, though perhaps they could be reworked to match the obvious embedded ChunkHeader usage. That might make the size calculation here a little more obvious. Perhaps move the waveTag field to another struct, and wrap it with the FormatChunk and data ChunkHeader. I am somewhat hesitant to deviate from documentation though. Thoughts?

Reply:
I agree it is difficult to read. I think I would prefer modifying the structs somehow to make it easier to read/understand. I'm not sure what that would look like right now though.

Reply:
Yes, lets defer. If we end up writing a Wave file class, that would be a good time to revisit this.